components/openapi/src/main/scala/pl/touk/nussknacker/openapi/http/backend/LoggingAndCollectingSttpBackend.scala

```diff
@@ -3,34 +3,31 @@ package pl.touk.nussknacker.openapi.http.backend
 import com.typesafe.scalalogging.Logger
 import org.slf4j.LoggerFactory
 import pl.touk.nussknacker.engine.api.test.InvocationCollectors.{CollectableAction, ServiceInvocationCollector, TransmissionNames}
-import pl.touk.nussknacker.openapi.http.backend.LoggingAndCollectingSttpBackend.HttpBackend
-import sttp.client.monad.MonadError
-import sttp.client.ws.WebSocketResponse
-import sttp.client._
+import sttp.capabilities.Effect
+import sttp.client3._
+import sttp.model.ResponseMetadata
 
 import java.nio.charset.StandardCharsets
 import java.util.UUID
 import scala.concurrent.{ExecutionContext, Future}
 
 object LoggingAndCollectingSttpBackend {
 
-  type HttpBackend = SttpBackend[Future, Nothing, Nothing]
-
   private case class WithStringResponseBody[T](original: T, stringResponseBody: () => Option[String])
 
   private val transmissionNames = TransmissionNames("request", "response")
 
 }
 
-class LoggingAndCollectingSttpBackend(delegate: HttpBackend, baseLoggerName: String)
-                                     (implicit ec: ExecutionContext, serviceInvocationCollector: ServiceInvocationCollector) extends HttpBackend {
+class LoggingAndCollectingSttpBackend[+P](delegate: SttpBackend[Future, P], baseLoggerName: String)
+                                     (implicit ec: ExecutionContext, serviceInvocationCollector: ServiceInvocationCollector) extends DelegateSttpBackend[Future, P](delegate) {
 
   import LoggingAndCollectingSttpBackend._
 
   private val requestLogger = Logger(LoggerFactory.getLogger(s"$baseLoggerName.request"))
   private val responseLogger = Logger(LoggerFactory.getLogger(s"$baseLoggerName.response"))
 
-  override def send[T](request: Request[T, Nothing]): Future[Response[T]] = {
+  override def send[T, R >: P with Effect[Future]](request: Request[T, R]): Future[Response[T]] = {
```

Comment:
Wow. Can't we fix this R generic type in class declaration? It looks like our usages of this class are simple - we don't need to change effect type for each invocation of send

Reply:
Then we wouldn't be able to use `DelegateSttpBackend`, which implements `close` and `responseMonad` for us

Reply:
I see now, o let's stay that way